discord/ui/section.py

```diff
@@ -0,0 +1,238 @@
+
+from .item import Item
+from .text_display import TextDisplay
+from ..enums import ComponentType
+from ..utils import MISSING, get as _utils_get
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .view import LayoutView
+    from ..components import SectionComponent
+
+V = TypeVar('V', bound='LayoutView', covariant=True)
+
+__all__ = ('Section',)
+
+
+class Section(Item[V]):
+    """Represents a UI section.
+
+    This is a top-level layout component that can only be used on :class:`LayoutView`
+
+    .. versionadded:: 2.6
+
+    Parameters
+    ----------
+    children: List[Union[:class:`str`, :class:`TextDisplay`]]
+        The text displays of this section. Up to 3.
+    accessory: :class:`Item`
+        The section accessory.
+    row: Optional[:class:`int`]
+        The relative row this section belongs to. By default
+        items are arranged automatically into those rows. If you'd
+        like to control the relative positioning of the row then
+        passing an index is advised. For example, row=1 will show
+        up before row=2. Defaults to ``None``, which is automatic
+        ordering. The row number must be between 0 and 9 (i.e. zero indexed)
+    id: Optional[:class:`int`]
+        The ID of this component. This must be unique across the view.
+    """
+
+    __discord_ui_section__: ClassVar[bool] = True
+    __discord_ui_update_view__: ClassVar[bool] = True
+
+    __slots__ = (
+        '_children',
+        'accessory',
+    )
+
+    def __init__(
+        self,
+        *children: Union[Item[V], str],
+        accessory: Item[V],
+        row: Optional[int] = None,
+        id: Optional[int] = None,
+    ) -> None:
+        super().__init__()
+        self._children: List[Item[V]] = []
+        if children is not MISSING:
+            if len(children) > 3:
+                raise ValueError('maximum number of children exceeded')
+            self._children.extend(
+                [c if isinstance(c, Item) else TextDisplay(c) for c in children],
+            )
+        self.accessory: Item[V] = accessory
+        self.row = row
+        self.id = id
+
+    @property
+    def type(self) -> Literal[ComponentType.section]:
+        return ComponentType.section
+
+    @property
+    def children(self) -> List[Item[V]]:
+        """List[:class:`Item`]: The list of children attached to this section."""
+        return self._children.copy()
+
+    @property
+    def width(self):
+        return 5
+
+    def _is_v2(self) -> bool:
+        return True
+
+    # Accessory can be a button, and thus it can have a callback so, maybe
+    # allow for section to be dispatchable and make the callback func
+    # be accessory component callback, only called if accessory is
+    # dispatchable?
+    def is_dispatchable(self) -> bool:
+        return self.accessory.is_dispatchable()
+
+    def walk_children(self) -> Generator[Item[V], Any, None]:
```

Comment:
Generators that only yield and dont utilize send values should have a send type of None, not Any
```suggestion
    def walk_children(self) -> Generator[Item[V], None, None]:
```